bundles/binding/org.openhab.binding.maxcube/src/main/java/org/openhab/binding/maxcube/internal/MaxCubeBinding.java

```diff
@@ -186,10 +199,14 @@ public synchronized void execute() {
 					cont = false;
 					continue;
 				}
-
-				Message message;
+				
+				Message message = null;
 				try {
-					message = processRawMessage(raw);
+					if (this.messageProcessor.addReceivedLine(raw)) {
```

Comment:
I was thinking about it, too.
In this way it was a bit easyer to implement AND it reduces the code, using the messageProcessor. 
It is a philosophical question...
If you really want me to introduce the messageProcessor.isMsgAvailable() method, I will implement it.